overlord/snapstate/snapstate_test.go

```diff
@@ -797,6 +797,67 @@ func (s snapmgrTestSuite) TestInstallFailsOnBusySnap(c *C) {
 	// And observe that we cannot refresh because the snap is busy.
 	_, err := snapstate.DoInstall(s.state, snapst, snapsup, 0, "")
 	c.Assert(err, ErrorMatches, `snap "some-snap" has running apps \(app\)`)
+
+	// The state records the time of the failed refresh operation.
+	err = snapstate.Get(s.state, "some-snap", snapst)
+	c.Assert(err, IsNil)
+	c.Check(snapst.RefreshInhibitedTime, NotNil)
+}
+
+func (s snapmgrTestSuite) TestInstallDespiteBusySnap(c *C) {
+	s.state.Lock()
+	defer s.state.Unlock()
+
+	// With the refresh-app-awareness feature enabled.
+	tr := config.NewTransaction(s.state)
+	tr.Set("core", "experimental.refresh-app-awareness", true)
+	tr.Commit()
+
+	// With a snap state indicating a snap is already installed and it failed
+	// to refresh over a week ago.
+	var longAgo = time.Now().Add(-time.Hour * 24 * 8)
+	// NOTE: Use rounding to a second granularity to
```

Comment:
the issue here is actually not quite the precision,  the issues are related to two things:

* the monotonic clock extra information,  `Round` removes it
* the time zone

using `time.Now().UTC().Add(-time.Hour * 24 * 8)`  and the `Round`  should solve both.